md4c/md4c.c

```diff
@@ -3630,6 +3649,36 @@ md_analyze_tilde(MD_CTX* ctx, int mark_index)
     }
 }
 
+static void
+md_analyze_dollar(MD_CTX* ctx, int mark_index)
+{
+    /* This should mimic the way inline equations work in LaTeX, so there
+     * can only ever be one item in the chain (i.e. the dollars can't be
+     * nested). This is basically the same as the md_analyze_tilde function,
+     * except that we require matching openers and closers to be of the same
+     * length.
+     *
+     * E.g.: $abc$$def$$ => abc (display equation) def (end equation) */
+    if(DOLLAR_OPENERS.head >= 0) {
+        /* If the potential closer has a non-matching number of $, discard */
+        MD_MARK open = ctx->marks[DOLLAR_OPENERS.head];
```

Comment:
Could you please use it via pointer instead of copying the whole structure? 
```
MD_MARK* opener = &ctx->marks[DOLLAR_OPENERS.head];
```

(Yes, I know it is small. But older C compilers could reject it.)

Reply:
Sure, that's no problem, see new commit. Out of curiosity, why would an older compiler reject it,  is it not valid C99?

Reply:
MD4C is now used by (dev branch of) Qt, and they also need to build for some embedded platform(s). Those often may use very ancient versions of gcc or even some proprietary compilers.

See #71.

Reply:
And BTW, 

AFAIK MSVC also does not support fully C99. It only knows some of its features (as extensions to some older standard, if they really adhere some at all). Microsoft repeatedly says they do care just about C++.

So I tend to be quite defensive.

Reply:
Fair. I wasn't aware of that. (I rarely use anything Microsoft)